jfix-zookeeper/src/main/kotlin/ru/fix/zookeeper/instance/registry/ServiceInstanceIdRegistry.kt

```
@@ -0,0 +1,196 @@
+        profiler: Profiler
+) : AutoCloseable {
+    lateinit var lockIdentity: LockIdentity
+        private set
+    lateinit var instanceIdPath: String
+        private set
+    lateinit var instanceId: String
+        private set
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(ServiceInstanceIdRegistry::class.java)
+    }
+
+    private val lockManager: PersistentExpiringLockManager = PersistentExpiringLockManager(
+            curatorFramework,
+            DynamicProperty.of(
+                    PersistentExpiringLockManagerConfig(
+                            reservationPeriod = config.disconnectTimeout.dividedBy(2),
+                            expirationPeriod = config.disconnectTimeout.dividedBy(3),
+                            lockProlongationInterval = config.disconnectTimeout.dividedBy(4),
+                            acquiringTimeout = Duration.ofSeconds(1)
+                    )
+            ),
+            profiler
+    )
+
+    init {
+        curatorFramework
+                .connectionStateListenable
+                .addListener(
+                        ConnectionStateListener { _, newState ->
+                            when (newState) {
+                                ConnectionState.RECONNECTED -> reconnect()
+                                else -> Unit
+                            }
+                        })
+        initServiceRegistrationPath()
+        initInstanceId()
+    }
+
+    private fun initServiceRegistrationPath() {
+        try {
+            curatorFramework.create()
+                    .creatingParentsIfNeeded()
+                    .forPath(config.serviceRegistrationPath, byteArrayOf())
+        } catch (e: KeeperException.NodeExistsException) {
+            logger.debug("Node ${config.serviceRegistrationPath} is already initialized", e)
+        } catch (e: Exception) {
+            logger.error("Illegal state when create path: ${config.serviceRegistrationPath}", e)
```

Why you are hiding Exception and allowing user to thing that ServiceInstanceRegistry correctly initialized, while it is not?

---

throw on error